src/base/globals.lua

```diff
@@ -74,13 +74,24 @@
 ---
 
 	premake.override(_G, "require", function(base, modname, versions)
-		local result, mod = pcall(base,modname)
+        local result, resmod = pcall(premake.moduleresolver.resolveModule, modname)
```

Comment:
resmod stands for _res_olved _mod_ule but maybe it's too clever, I'll update it with something more readable.